### DISCUSSION TITLE
Storage reached its minimum free disk threshold, but my disk space only used 38%.

### DISCUSSION THREAD

**Discussion by BaiJing-LR** [minio/minio#13282] [Q&A]
## Current Behavior

API: SYSTEM(bucket=miniotest, object=2:1235740:15165408606258138108)
Time: 22:38:21 EDT 09/22/2021
DeploymentID: 1e325925-fbed-45e9-b1c1-f2a7e13bbe91
Error: disk path full (cmd.StorageErr)
      16: cmd/erasure-healing.go:546:cmd.erasureObjects.healObject()
      15: cmd/erasure-healing.go:932:cmd.erasureObjects.HealObject()
      14: cmd/erasure-sets.go:1347:cmd.(*erasureSets).HealObject()
      13: cmd/erasure-server-pool.go:1738:cmd.(*erasureServerPools).HealObject()
      12: cmd/data-scanner.go:875:cmd.(*scannerItem).applyHealing()
      11: cmd/data-scanner.go:978:cmd.(*scannerItem).applyActions()
      10: cmd/xl-storage.go:497:cmd.(*xlStorage).NSScanner.func1()
       9: cmd/data-scanner.go:472:cmd.(*folderScanner).scanFolder.func1()
       8: cmd/os-readdir_unix.go:172:cmd.readDirFn()
       7: cmd/data-scanner.go:405:cmd.(*folderScanner).scanFolder()
       6: cmd/data-scanner.go:527:cmd.(*folderScanner).scanFolder.func2()
       5: cmd/data-scanner.go:573:cmd.(*folderScanner).scanFolder()
       4: cmd/data-scanner.go:312:cmd.scanDataFolder()
       3: cmd/xl-storage.go:467:cmd.(*xlStorage).NSScanner()
       2: cmd/storage-rest-server.go:203:cmd.(*storageRESTServer).NSScannerHandler()
       1: net/http/server.go:2046:http.HandlerFunc.ServeHTTP()

API: SYSTEM()
Time: 22:38:21 EDT 09/22/2021
DeploymentID: 1e325925-fbed-45e9-b1c1-f2a7e13bbe91
Error: Storage reached its minimum free disk threshold. (cmd.StorageFull)
      12: cmd/data-scanner.go:877:cmd.(*scannerItem).applyHealing()
      11: cmd/data-scanner.go:978:cmd.(*scannerItem).applyActions()
      10: cmd/xl-storage.go:497:cmd.(*xlStorage).NSScanner.func1()
       9: cmd/data-scanner.go:472:cmd.(*folderScanner).scanFolder.func1()
       8: cmd/os-readdir_unix.go:172:cmd.readDirFn()
       7: cmd/data-scanner.go:405:cmd.(*folderScanner).scanFolder()
       6: cmd/data-scanner.go:527:cmd.(*folderScanner).scanFolder.func2()
       5: cmd/data-scanner.go:573:cmd.(*folderScanner).scanFolder()
       4: cmd/data-scanner.go:312:cmd.scanDataFolder()
       3: cmd/xl-storage.go:467:cmd.(*xlStorage).NSScanner()
       2: cmd/storage-rest-server.go:203:cmd.(*storageRESTServer).NSScannerHandler()
       1: net/http/server.go:2046:http.HandlerFunc.ServeHTTP()

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio --version`):

`minio version RELEASE.2021-09-18T18-09-59Z`

* Server setup and configuration:

`minio server http://192.168.2.25{0...3}/home/minio/disk0 --console-address :9001`

![image](https://user-images.githubusercontent.com/90737547/134447153-d4f52b65-79a7-4bc9-af75-0d33113abf0e.png)

* Operating System and version (`uname -a`):

`Linux localhost.localdomain 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by harshavardhana** [comment]
Is your filesystem ext4? @BaiJing-LR

**Reply by BaiJing-LR** [reply]
Yes, what is the impact ?  @harshavardhana

**Reply by harshavardhana** [reply]
ext4 has 64k directory limit per-prefix - you need to use XFS @BaiJing-LR  - this is ext4 limitation.

**Reply by BaiJing-LR** [reply]
I see, Thinks.
